enclave-runner/src/usercalls/mod.rs

```diff
@@ -346,36 +381,204 @@ impl EnclaveState {
             fds: Mutex::new(fds),
             last_fd,
             exiting: AtomicBool::new(false),
-            usercall_ext : usercall_ext,
+            usercall_ext,
         })
     }
 
+    fn create_worker_threads(num_of_worker_threads: usize,
+                             work_receiver: crossbeam::crossbeam_channel::Receiver<Work>,
+                             io_queue_send: mpsc::Sender<UsercallSendData>
+    ) -> Vec<JoinHandle<()>> {
+        let mut thread_handles = vec![];
+        for _ in 0..num_of_worker_threads {
+            let work_receiver = work_receiver.clone();
+            let io_queue_send = io_queue_send.clone();
+
+            thread_handles.push(thread::spawn(move || {
+                while let Ok(work) = work_receiver.recv() {
+                    work.do_work(&io_queue_send);
+                }
+            }));
+        }
+        return thread_handles;
+    }
+
+    fn  syscall_loop(io_queue_receive: mpsc::Receiver<UsercallSendData>, enclave: Arc<EnclaveState>,
+                     work_sender: crossbeam::crossbeam_channel::Sender<Work>
+    ) -> StdResult<(u64, u64), EnclaveAbort<EnclavePanic>> {
+        while let Some((coresult, mut state, buf)) = io_queue_receive.recv().ok() {
+            let mut erased_tcs = None;
```

Comment:
I don't think you need to save this in a variable. You can always just put it back on the queue then and there instead of assigning it.